nepi/main/tests/test_views.py

```diff
@@ -630,6 +631,138 @@ def test_create_school(self):
+        response = self.client.get('/join_group/')
+        self.assertEquals(response.status_code, 405)
+
+        # invalid data
+        response = self.client.post('/join_group/', {})
+        self.assertEquals(response.status_code, 404)
+
+    def test_join_group(self):
+        self.assertEquals(self.student.profile.group.count(), 0)
+
+        self.client.login(username=self.student.username, password="test")
+        response = self.client.post('/join_group/', {'group': self.group.id})
+        self.assertEquals(response.status_code, 302)
+
+        self.assertEquals(self.student.profile.group.count(), 1)
+
+
+class TestDeleteGroupView(TestCase):
+    def setUp(self):
+        self.student = StudentProfileFactory().user
+        self.creator = TeacherProfileFactory().user
+        self.teacher = TeacherProfileFactory().user
+
+        self.group = SchoolGroupFactory(creator=self.creator)
+
+    def test_access(self):
+        # not logged in
+        response = self.client.post('/delete_group/', {})
+        self.assertEquals(response.status_code, 302)
+
+        # not authorized
+        self.client.login(username=self.student.username, password="test")
+        response = self.client.post('/delete_group/', {'group': self.group.id})
+        self.assertEquals(response.status_code, 403)
+
+        # not authorized
+        self.client.login(username=self.teacher.username, password="test")
+        response = self.client.post('/delete_group/', {'group': self.group.id},
+                                    HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEquals(response.status_code, 403)
+
+        # invalid method
+        self.client.login(username=self.creator.username, password="test")
+        response = self.client.get('/delete_group/')
+        self.assertEquals(response.status_code, 405)
+
+        # not ajax
+        self.client.login(username=self.creator.username, password="test")
+        response = self.client.post('/delete_group/', {})
+        self.assertEquals(response.status_code, 405)
+
+        # required data not available
+        self.client.login(username=self.creator.username, password="test")
+        response = self.client.post('/delete_group/', {},
+                                    HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEquals(response.status_code, 404)
+
+    def test_post(self):
+        self.client.login(username=self.creator.username, password="test")
+        response = self.client.post('/delete_group/', {'group': self.group.id},
+                                    HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.assertEquals(response.status_code, 200)
+        self.assertTrue(loads(response.content)['success'])
+
+        try:
+            Group.objects.get(id=self.group.id)
+            self.fail('expected group to be deleted')
+        except Group.DoesNotExist:
+            pass  # expected
```

Comment:
If this is the expected case, this assertion could also be written using [assertRaises](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises), but that works.